src/Map.jsx

```diff
@@ -55,8 +64,19 @@ class Map extends Component {
           <NavigationControl onViewportChange={onViewportChange} />
         </div>
 
+        {/* React Map GL Popup component displays vehicle ID & heading info */}
+        <Popup
+          longitude={this.state.coordinates[0]}
+          latitude={this.state.coordinates[1]}
+        >
+          <div>
+            <p>ID: {this.state.info.vid}</p>
+            <p>Heading: {this.state.info.heading}</p>
+          </div>
+        </Popup>
+
         {/* Routes component returns DeckGL component with routes and markers layer */}
-        <Routes state={this.props.state} viewport={viewport} />
+        <Routes onMarkerClick={this.onMarkerClick} state={this.props.state} viewport={viewport} />
```

Comment:
for cleaner syntax, you can use ES6 bindings here. you can do `onMarkerClick={() => this.onMarkerClick()}` instead, so then you'd no longer have to bind it in line 42

Reply:
I didn't know that usage, alright